### DISCUSSION TITLE
Shutters with Relay1 down and Relay2 up

### DISCUSSION THREAD

**Discussion by tomm1ed** [arendst/Tasmota#15429] [Q&A] [answered]
### PROBLEM DESCRIPTION
I have flashed a couple of Sonoff T0 2 Gang switches with Tasmota. I have Relay1 connected to Down and Relay2 to Up for my shutter, because I want to use the Left button for down and the right for up.
When configured with Shutter mode 1 - Normal Operation however the default is Relay1: UP/OFF, Relay2: DOWN/OFF meaning that both the Tasmota interface and the connected Home Assistant always shows the wrong status (Closed when Open and vice-versa)

I've tried corrrecting it with ShutterInvertWebButtons 1 / ShutterInvert 1 but can't seem to get it to work.

Config:
ledstate 0
SetOption80 1
Interlock 1,2
Interlock ON
ShutterRelay1 1
ShutterOpenDuration 30
ShutterCloseDuration 31
teleperiod 60

### REQUESTED INFORMATION
_Make sure your have performed every step and checked the applicable boxes before submitting your issue. Thank you!_

- [X] Read the [Contributing Guide and Policy](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md) and [the Code of Conduct](https://github.com/arendst/Tasmota/blob/development/CODE_OF_CONDUCT.md)
- [X] Searched the problem in [issues](https://github.com/arendst/Tasmota/issues)
- [X] Searched the problem in [discussions](https://github.com/arendst/Tasmota/discussions)
- [X] Searched the problem in the [docs](https://tasmota.github.io/docs/FAQ)
- [X] Searched the problem in the [chat](https://discord.gg/Ks2Kzd4)
- [X] Device used (e.g., Sonoff Basic): SonOff T0 2 Channel
- [X] Tasmota binary firmware version number used: 9.1.0, 9.3.1, 11.0.0
  - [X] Pre-compiled
  - [ ] Self-compiled
- [X] Flashing tools used: Tasmotizer
- [ ] Provide the output of command: `Backlog Template; Module; GPIO 255`:
```lua
  Configuration output here:
11:20:48.703 MQT: stat/tasmota_773757/RESULT = {"NAME":"Generic","GPIO":[32,1,1,1,0,225,33,0,224,320,0,0,0,0],"FLAG":0,"BASE":29}
11:20:48.917 MQT: stat/tasmota_773757/RESULT = {"Module":{"29":"Sonoff T1 2CH"}}
11:20:49.170 MQT: stat/tasmota_773757/RESULT = {"GPIO0":{"32":"Button1"},"GPIO1":{"0":"None"},"GPIO2":{"0":"None"},"GPIO3":{"0":"None"},"GPIO4":{"0":"None"},"GPIO5":{"225":"Relay2"},"GPIO9":{"33":"Button2"},"GPIO10":{"0":"None"},"GPIO12":{"224":"Relay1"},"GPIO13":{"320":"Led_i1"},"GPIO14":{"0":"None"},"GPIO15":{"0":"None"},"GPIO16":{"0":"None"},"GPIO17":{"0":"None"}}

```
- [ ] If using rules, provide the output of this command: `Backlog Rule1; Rule2; Rule3`:
```lua
  Rules output here:
11:21:23.817 MQT: stat/tasmota_773757/RESULT = {"Rule1":"OFF","Once":"OFF","StopOnError":"OFF","Length":0,"Free":511,"Rules":""}
11:21:24.032 MQT: stat/tasmota_773757/RESULT = {"Rule2":"OFF","Once":"OFF","StopOnError":"OFF","Length":0,"Free":511,"Rules":""}
11:21:24.275 RUL: Stored uncompressed, would compress from 1 to 2 (-0%)
11:21:24.285 MQT: stat/tasmota_773757/RESULT = {"Rule3":"OFF","Once":"OFF","StopOnError":"OFF","Length":1,"Free":510,"Rules":"`"}

```
- [ ] Provide the output of this command: `Status 0`:
```lua
  STATUS 0 output here:
11:26:55.533 CMD: Status 0
11:26:55.541 MQT: stat/tasmota_773757/STATUS = {"Status":{"Module":29,"DeviceName":"Rolluik-maya","FriendlyName":["Tasmota","Tasmota2"],"Topic":"tasmota_773757","ButtonTopic":"0","Power":0,"PowerOnState":3,"LedState":0,"LedMask":"FFFF","SaveData":1,"SaveState":1,"SwitchTopic":"0","SwitchMode":[0,0,0,0,0,0,0,0],"ButtonRetain":0,"SwitchRetain":0,"SensorRetain":0,"PowerRetain":0,"InfoRetain":0,"StateRetain":0}}
11:26:55.547 MQT: stat/tasmota_773757/STATUS1 = {"StatusPRM":{"Baudrate":115200,"SerialConfig":"8N1","GroupTopic":"tasmotas","OtaUrl":"http://ota.tasmota.com/tasmota/release/tasmota.bin.gz","RestartReason":"Software/System restart","Uptime":"0T00:49:03","StartupUTC":"2022-04-22T09:37:52","Sleep":50,"CfgHolder":4617,"BootCount":181,"BCResetTime":"2021-06-30T14:16:08","SaveCount":2096,"SaveAddress":"F4000"}}
11:26:55.554 MQT: stat/tasmota_773757/STATUS2 = {"StatusFWR":{"Version":"9.3.1(tasmota)","BuildDateTime":"2021-02-23T12:02:38","Boot":31,"Core":"2_7_4_9","SDK":"2.2.2-dev(38a443e)","CpuFrequency":80,"Hardware":"ESP8285","CR":"410/699"}}
11:26:55.559 MQT: stat/tasmota_773757/STATUS3 = {"StatusLOG":{"SerialLog":2,"WebLog":2,"MqttLog":0,"SysLog":0,"LogHost":"","LogPort":514,"SSId":["KT5",""],"TelePeriod":300,"Resolution":"558180C0","SetOption":["0000C009","2805C8000100060000005A0A000000000000","40000080","00006000","00000000"]}}
11:26:55.568 MQT: stat/tasmota_773757/STATUS4 = {"StatusMEM":{"ProgramSize":593,"Free":408,"Heap":24,"ProgramFlashSize":1024,"FlashSize":1024,"FlashChipId":"144051","FlashFrequency":40,"FlashMode":3,"Features":["00000809","8FDAC787","04368001","000000CF","010013C0","C000F981","00004004","00001000","00000000"],"Drivers":"1,2,3,4,5,6,7,8,9,10,12,16,18,19,20,21,22,24,26,27,29,30,35,37,45","Sensors":"1,2,3,4,5,6"}}
11:26:55.575 MQT: stat/tasmota_773757/STATUS5 = {"StatusNET":{"Hostname":"sonoff-rolluik-maya","IPAddress":"192.168.12.52","Gateway":"192.168.12.1","Subnetmask":"255.255.255.0","DNSServer":"192.168.12.1","Mac":"84:0D:8E:77:37:57","Webserver":2,"WifiConfig":4,"WifiPower":17.0}}
11:26:55.580 MQT: stat/tasmota_773757/STATUS6 = {"StatusMQT":{"MqttHost":"192.168.11.23","MqttPort":1883,"MqttClientMask":"DVES_%06X","MqttClient":"DVES_773757","MqttUser":"DVES_USER","MqttCount":1,"MAX_PACKET_SIZE":1200,"KEEPALIVE":30}}
11:26:55.589 MQT: stat/tasmota_773757/STATUS7 = {"StatusTIM":{"UTC":"2022-04-22T10:26:55","Local":"2022-04-22T11:26:55","StartDST":"2022-03-27T02:00:00","EndDST":"2022-10-30T03:00:00","Timezone":"+01:00","Sunrise":"05:45","Sunset":"19:51"}}
11:26:55.597 MQT: stat/tasmota_773757/STATUS10 = {"StatusSNS":{"Time":"2022-04-22T11:26:55","Shutter1":{"Position":0,"Direction":0,"Target":0}}}
11:26:55.604 MQT: stat/tasmota_773757/STATUS11 = {"StatusSTS":{"Time":"2022-04-22T11:26:55","Uptime":"0T00:49:03","UptimeSec":2943,"Heap":25,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":1,"POWER1":"OFF","POWER2":"OFF","Wifi":{"AP":1,"SSId":"KT5","BSSId":"80:8D:B7:81:4B:F2","Channel":11,"RSSI":44,"Signal":-78,"LinkCount":1,"Downtime":"0T00:00:03"}}}
11:26:55.611 MQT: stat/tasmota_773757/STATUS13 = {"StatusSHT":{"SHT0":{"Relay1":1,"Relay2":2,"Open":210,"Close":210,"50perc":50,"Delay":0,"Opt":"0000","Calib":[300,500,700,900,1000],"Mode":"0"}}}

```
- [ ] Set `weblog` to 4 and then, when you experience your issue, provide the output of the Console log:
```lua
  Console output here:


```

### TO REPRODUCE
_Steps to reproduce the behavior:_

Connect Relay1 to Shutter Down
Connect Relay2 to Shutter Up

### EXPECTED BEHAVIOUR
_A clear and concise description of what you expected to happen._
Being able to let Relay1 control Down in stead of Up
Being able to let Relay2 control Up in stead of Down
Correct representation of shutter status (Down when down and Up when up)

### SCREENSHOTS
_If applicable, add screenshots to help explain your problem._

### ADDITIONAL CONTEXT
_Add any other context about the problem here._

**(Please, remember to close the issue when the problem has been addressed)**

**Comment by tomm1ed** [comment, marked as answer]
Is the fact that this is placed in Discussion a sign that this is expected behavior? (and also that I shouldn't expect this feature to be added?)

**Reply by sfromis** [reply]
No, it's a sign that what looks like a question of how to use things was wrongly created as an "issue", and hence moved to a more appropriate place by an administrator, to have better chances of getting attention.

**Reply by tomm1ed** [reply]
Ok no reply so my conclusion appears to be correct that this is just not possible with tasmota so it would seem it now is more of a feature request than anything else…

For now I will just switch the wires and teach everyone in the house the controls are now upside down because defining up and down in Tasmota apparently is not possible.

**Reply by barbudor** [reply]
Have you tried the obvious thing to switch the index of relays on the GPIOs?

**Reply by tomm1ed** [reply]
I am not sure what you mean by that

**Reply by barbudor** [reply]
If tasmota use Relay1 for Up and Relay2 for Down, just adapt the template so that Relay1 is on the GPIO that drives the Up relay and Relay2 is on the GPIO that drives the Down relay
So just swap the index 1 ans index 2 in the template

**Reply by tomm1ed** [reply]
Thanks for the idea. I just tested this and it makes absolutely no difference :(

**Reply by barbudor** [reply]
That makes no sense
What is now the result of `Backlog Template; Module; GPIO 255` ?

**Reply by tomm1ed** [reply]
With Relay 1 and 2 swapped:
[
![](https://tweakers.net/fotoalbum/image/MKK7Y7lHJMIEc3HYei7PFIC1.png)
](url)
13:28:02.985 MQT: stat/tasmota_773767/RESULT = {"NAME":"Generic","GPIO":[32,1,1,1,0,224,33,0,225,320,0,0,0,0],"FLAG":0,"BASE":29}
13:28:03.204 MQT: stat/tasmota_773767/RESULT = {"Module":{"29":"Sonoff T1 2CH"}}
13:28:03.456 MQT: stat/tasmota_773767/RESULT = {"GPIO0":{"32":"Button1"},"GPIO1":{"0":"None"},"GPIO2":{"0":"None"},"GPIO3":{"0":"None"},"GPIO4":{"0":"None"},"GPIO5":{"225":"Relay2"},"GPIO9":{"33":"Button2"},"GPIO10":{"0":"None"},"GPIO12":{"224":"Relay1"},"GPIO13":{"320":"Led_i1"},"GPIO14":{"0":"None"},"GPIO15":{"0":"None"},"GPIO16":{"0":"None"},"GPIO17":{"0":"None"}}

With the normal setup:
[
![](https://tweakers.net/fotoalbum/image/BSrPrbFI8Zy6CAIRH21YIunc.png)
](url)

13:31:17.162 MQT: stat/tasmota_773767/RESULT = {"NAME":"Generic","GPIO":[32,1,1,1,0,225,33,0,224,320,0,0,0,0],"FLAG":0,"BASE":29}
13:31:17.380 MQT: stat/tasmota_773767/RESULT = {"Module":{"29":"Sonoff T1 2CH"}}
13:31:17.632 MQT: stat/tasmota_773767/RESULT = {"GPIO0":{"32":"Button1"},"GPIO1":{"0":"None"},"GPIO2":{"0":"None"},"GPIO3":{"0":"None"},"GPIO4":{"0":"None"},"GPIO5":{"225":"Relay2"},"GPIO9":{"33":"Button2"},"GPIO10":{"0":"None"},"GPIO12":{"224":"Relay1"},"GPIO13":{"320":"Led_i1"},"GPIO14":{"0":"None"},"GPIO15":{"0":"None"},"GPIO16":{"0":"None"},"GPIO17":{"0":"None"}}

**Reply by barbudor** [reply]
In both cases Relay2 is on GPIO5 and Relay1 is on GPIO12
It looks like the Module 29 override the GPIOs in Template, so the swapped case is not take into account

You should be able to use Module 0 instead in to use manually defined template
The only impact I see in the code could be on the Led which may show with the expected way. If this is the case you can change Led_i for Led.
But try 1st the swapped template with module 0

**Reply by tomm1ed** [reply]
Wow can't believe I missed that.
The representation of open and close is now correct but the buttons are also swapped so the left button (which I want to use to close the shutter) now opens it and vice-versa. Funnily enough the light goes on at the wrong button so the LED under the button seams to be hardwired to the Relay.

FYI I also tried swapping the button numbers but then the open/close status is not working again. I still think that it should be a bit more trivial to tell Tasmota which button is up and which is down but for now I'll just swap the relays and accept that up means down and down means up.

**Reply by barbudor** [reply]
You mean Physical Buttons (not Web buttons) ?
For Tasmota 
- Button 1 => Relay 1 => Up
- Button 2 => Relay 2 => Down

So it's up to you to chose which GPIO are on Button 1/2 and Relay 1/2
If you have successfully swapped the relays, you can also swapped the physical buttons the same

**Comment by tomm1ed** [comment]
They are touch buttons that light up when pressed.
![11AE912D-98B2-4951-B503-9BD953FEC3E4](https://user-images.githubusercontent.com/82563598/166895643-d6b606c9-33a8-40fc-90d2-5c03f0cd6e75.jpeg)

 It looks like the LEDs behind the buttons are hard wired to the relays (so the left LED lights up when relay1 is active and the right LED lights up when relay2 is active)
